content/docs/07-reference/01-ai-sdk-core/80-smooth-stream.mdx

```diff
@@ -42,14 +42,60 @@ const result = streamText({
     },
     {
       name: 'chunking',
-      type: '"word" | "line" | RegExp',
+      type: '"word" | "line" | RegExp | (buffer: string) => string | undefined | null',
       isOptional: true,
       description:
-        'Controls how the text is chunked for streaming. Use "word" to stream word by word (default), "line" to stream line by line, or provide a custom RegExp pattern for custom chunking.',
+        'Controls how the text is chunked for streaming. Use "word" to stream word by word (default), "line" to stream line by line, or provide a custom callback or RegExp pattern for custom chunking.',
     },
   ]}
 />
 
+#### Word chunking caveats with non-latin languages
+
+<Note>
+    The word based chunking **does not work well** with the following languages that do not delimit words with spaces:
+
+    For these languages the `word` option splits on the entire unicode range:
+
+    - Chinese - `\u4E00-\u9FFF`
+    - Japanese - `\u3040-\u309F` & `\u30A0-\u30FF`
+
+    For these languages you could pass your own language aware chunking function:
+
+    - Vietnamese
+    - Thai
+    - Javanese (Aksara Jawa)
+
+</Note>
+
+#### Regex based chunking
+
+To use regex based chunking, pass a `RegExp` to the `chunking` option.
+
+```ts
+// To split on underscores:
+smoothStream({
+  chunking: /_+/,
+});
```

Comment:
would be good to have one example with full match and to explain the option and implications

Reply:
good idea